app/models/bishop.rb

```diff
@@ -1,7 +1,10 @@
 class Bishop < ChessPiece
-  def valid_move?(coordinates)
+  def valid_move?(coordinates, skip_checkstate_check = false)
```

Comment:
I'm not super familiar, so ignore me if this seems off.

Where is `skip_checkstate_check` set to `true`?

In cases like this I'd usually call
`valid_move?(coordinates)` and then
`move_puts_king_in_check?(coordinates)`

Rather than having it in 1 method (if it's not happening every time).